cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -137,6 +137,7 @@ class HydratorPlusPlusTopPanelCtrl{
   }
   onSaveDraft() {
     this.HydratorPlusPlusConfigActions.saveAsDraft();
+    this.checkNameError();
```

Comment:
this should be checked first, and if it errors out, line 139 should not be called. Otherwise, the draft is being saved still with an empty name